scripts/utils/docs-content-tools/handlers.js

```diff
@@ -0,0 +1,157 @@
+ * @returns {String} The slug of the filepath
+ */
+const getSlugFromPath = (filepath) => {
+  const slug = filepath.match(/content(.*?).mdx/)[1];
+  if (!slug) {
+    console.error(
+      `<!> Could not find slug from path: ${filepath}, please check filepath`
+    );
+    process.exit(1);
+  }
+  return slug;
+};
+
+/**
+ * Reads a file and separates the content and frontmatter data
+ * @param {String} path - The full filepath of the file
+ * @returns {Object} Original path, slug from path, content, frontmatter data
+ */
+const separateData = (path) => {
+  const file = fs.readFileSync(path, 'utf8');
+  const slug = getSlugFromPath(path);
+  const separateData = frontmatterGH(file);
+  const { data: frontmatterData, content } = separateData;
+  if (!frontmatterData.redirects) {
+    frontmatterData.redirects = [];
+  }
+  return { path, content, frontmatterData, slug };
+};
+
+/**
+ * Joins the content and frontmatter data
+ * @param {Object} data - content, frontmatterData, path
+ * @returns {Object} Combined data and path
+ */
+const joinData = ({ content, frontmatterData, path }) => {
+  const data = frontmatterGH.stringify(content, frontmatterData, {
+    lineWidth: -1,
+  });
+
+  return { path, data };
+};
+
+/**
+ * Adds a redirect if the slug is not already in the redirects array
+ * @param {Object} data - slug, frontmatterData, rest
+ * @returns {Object} Edited frontmatterData, whether the redirect was added or skipped
+ */
+const addRedirect = ({ frontmatterData, slug, ...rest }) => {
+  if (frontmatterData.redirects.includes(slug)) {
+    return { frontmatterData, ...rest, skipped: slug };
+  }
+  return {
+    frontmatterData: {
+      ...frontmatterData,
+      redirects: [...frontmatterData.redirects, slug],
+    },
+    ...rest,
+    added: slug,
+  };
+};
+
+/**
+ * Removes a redirect if the slug is in the redirects array
+ * @param {Object} data - slug, frontmatterData, rest
+ * @returns {Object} Edited frontmatterData, whether the redirect was removed or skipped
+ */
+const removeRedirect = ({ frontmatterData, slug, ...rest }) => {
+  if (frontmatterData.redirects.includes(slug)) {
+    frontmatterData.redirects = frontmatterData.redirects.filter(
+      (path) => path !== slug
+    );
+    if (frontmatterData.redirects.length === 0) {
+      delete frontmatterData.redirects;
+    }
+    return {
+      frontmatterData,
+      ...rest,
+      removed: slug,
+    };
+  }
+  return { frontmatterData, ...rest, skipped: slug };
+};
+
+/**
+ * Globs an array of all mdx files in a directory + sub-directories
+ * @param {String} directory - The parent directory to search
+ * @returns {String[]} All mdx files in the directory and sub-directories
+ */
+const getFilePaths = (directory) => {
+  if (directory.length === 0) {
+    // if user did not supply paths, default to all
+    return glob.sync(`${__dirname}/../../../src/content/docs/**/*.mdx`);
+  }
+
+  return glob.sync(`${__dirname}/../../../src/content${directory}**/*.mdx`);
```

Comment:
Totally a style thing, not sure if this would work right.

```suggestion
  const subDir = directory.length ? directory : '/docs/';
  return glob.sync(`${__dirname}/../../../src/content${subDir}**/*.mdx`);
```